### PR TITLE
add first class rsh, better deal with args for proxy cmds like exec, rsh

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins-client-plugin/issues/79

@openshift/sig-developer-experience fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

File: src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java [hunk 1]
```diff
@@ -111,6 +106,14 @@ private boolean hasArg(List<String> args, String... argsToFind) {
...
             }
         }
+        
+        cmd.add(verb);
+
+        cmd.addAll(toStringArray(verbArgs));
+
+        cmd.addAll(toStringArray(userArgs));
+
+        cmd.addAll(toStringArray(options));
```
Comment (bparees) [inline src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java]: this is just ensuring that the "--" arg does not come before the "--loglevel" or other args being added?

Comment (gabemontero) [reply src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java]: ah ... i should update the comment, forgot that the literal `--` sequence could be a valid arg

what I was trying to convey was that things like --server, --namespace, --token, --loglevel in the wrong position would confuse `oc rsh` ... that is what was causing the problem .... moving those args prior to the verb works generically, and protects `oc rsh`

Comment (bparees) [reply src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java]: got it, yeah a clarifying comment would help, otherwise lgtm.

Comment (gabemontero) [reply src/main/java/com/openshift/jenkins/plugins/util/ClientCommandBuilder.java]: comment clarified

**Comment by gabemontero** [comment]
[merge]

**Pull request merged**